### PR TITLE
qemu_img_convert_from_vdi: new case

### PR DESCRIPTION
**Pull request by hellohellenmao**
1. Convert from vdi os image
2. Boot up from the target image

Signed-off-by: Tingting Mao <timao@redhat.com>

ID: 2112262

### CODE REVIEW

**Comment by hellohellenmao** [comment]
(1/1) Host_RHEL.m9.u1.qcow2.ide.up.virtio_net.Guest.RHEL.8.7.0.x86_64.io-github-autotest-qemu.qemu_img_convert_from_vdi: PASS (1327.94 s)

**Comment by hellohellenmao** [comment]
@XueqiangWei @zhencliu Could you please help to take a review here? Thanks.

File: qemu/tests/cfg/qemu_img_convert_from_vdi.cfg [hunk 1]
```diff
@@ -0,0 +1,19 @@
+- qemu_img_convert_from_vdi:
+    only RHEL
+    only i440fx
+    no Host_RHEL.m6 Host_RHEL.m7 Host_RHEL.m8
+    required_qemu = [6.1.0-7, )
+    virt_test_type = qemu
+    type = qemu_img_convert_from_vdi
+    start_vm = no
+    kill_vm = yes
+    vdi_image_address = "http://fileshare.englab.nay.redhat.com/pub/section2/kvm/timao/vdi_image/rhel850.vdi"
```
Comment (zhencliu) [inline qemu/tests/cfg/qemu_img_convert_from_vdi.cfg:10]: never attach 'private info' in upstream

File: qemu/tests/cfg/qemu_img_convert_from_vdi.cfg [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    rm_testfile_cmd = "rm -f %s"
+    images = "rhel850"
+    image_name_rhel850 = images/rhel850
+    image_format_rhel850 = vdi
+    convert_target = "convert"
+    image_name_convert = images/from_vdi
+    remove_image = yes
+    image_format_convert = qcow2
```
Comment (zhencliu) [inline qemu/tests/cfg/qemu_img_convert_from_vdi.cfg:18]: seems we dont need to set this, for the format is set in py

File: qemu/tests/qemu_img_convert_from_vdi.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    for format in ("qcow2", "raw"):
+        params["image_format_convert"] = format
+        # Convert the source image to target
+        source = QemuImg(params.object_params(src_image), img_dir, src_image)
+        fail_on((process.CmdError,))(source.convert)(source.params, img_dir)
+
+        # Boot from the target image
+        img_utils.boot_vm_with_images(test, params, env, (tgt_image,))
+
+    params["images"] += " " + tgt_image
```
Comment (zhencliu) [inline qemu/tests/qemu_img_convert_from_vdi.py:39]: only one target image can be removed, but we create two above, one is in raw format and the other is in qcow2 format

Comment (zhencliu) [reply qemu/tests/qemu_img_convert_from_vdi.py:39]: so what about
try:
  boot_vm_with_images
finally:
  target.remove()

File: qemu/tests/cfg/qemu_img_convert_from_vdi.cfg [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    only RHEL
+    only i440fx
+    no Host_RHEL.m6 Host_RHEL.m7 Host_RHEL.m8
+    required_qemu = [6.1.0-7, )
+    virt_test_type = qemu
+    type = qemu_img_convert_from_vdi
+    start_vm = no
+    kill_vm = yes
+    vdi_image_address = "http://fileshare.englab.nay.redhat.com/pub/section2/kvm/timao/vdi_image/rhel850.vdi"
+    rm_testfile_cmd = "rm -f %s"
```
Comment (zhencliu) [inline qemu/tests/cfg/qemu_img_convert_from_vdi.cfg:11]: not used?

File: qemu/tests/qemu_img_convert_from_vdi.py (hunk 3 above)
Comment (hellohellenmao) [reply qemu/tests/qemu_img_convert_from_vdi.py]: Updated. Thanks.

File: qemu/tests/qemu_img_convert_from_vdi.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    process.system(vdi_image_download_cmd)
+
+    for format in ("qcow2", "raw"):
+        params["image_format_convert"] = format
+        # Convert the source image to target
+        source = QemuImg(params.object_params(src_image), img_dir, src_image)
+        fail_on((process.CmdError,))(source.convert)(source.params, img_dir)
+
+        # Boot from the target image
+        img_utils.boot_vm_with_images(test, params, env, (tgt_image,))
```
[the remark this answers is not in the document]
Comment (hellohellenmao) [reply qemu/tests/qemu_img_convert_from_vdi.py]: No need here.

File: qemu/tests/cfg/qemu_img_convert_from_vdi.cfg (hunk 1 above)
Comment (hellohellenmao) [reply qemu/tests/cfg/qemu_img_convert_from_vdi.cfg]: Thanks, have added the info to the test loop.

**Comment by hellohellenmao** [comment]
The results for VDI test loop:
 
(1/4) Host_RHEL.m9.u1.qcow2.ide.up.virtio_net.Guest.RHEL.8.7.0.x86_64.io-github-autotest-qemu.vdi_image_convert.to_raw: STARTED
 (1/4) Host_RHEL.m9.u1.qcow2.ide.up.virtio_net.Guest.RHEL.8.7.0.x86_64.io-github-autotest-qemu.vdi_image_convert.to_raw: PASS (3.34 s)
 (2/4) Host_RHEL.m9.u1.qcow2.ide.up.virtio_net.Guest.RHEL.8.7.0.x86_64.io-github-autotest-qemu.vdi_image_convert.to_vdi: STARTED
 (2/4) Host_RHEL.m9.u1.qcow2.ide.up.virtio_net.Guest.RHEL.8.7.0.x86_64.io-github-autotest-qemu.vdi_image_convert.to_vdi: PASS (3.34 s)
 (3/4) Host_RHEL.m9.u1.qcow2.ide.up.virtio_net.Guest.RHEL.8.7.0.x86_64.io-github-autotest-qemu.vdi_image_create: STARTED
 (3/4) Host_RHEL.m9.u1.qcow2.ide.up.virtio_net.Guest.RHEL.8.7.0.x86_64.io-github-autotest-qemu.vdi_image_create: PASS (1.67 s)
 (4/4) Host_RHEL.m9.u1.qcow2.ide.up.virtio_net.Guest.RHEL.8.7.0.x86_64.io-github-autotest-qemu.qemu_img_convert_from_vdi: STARTED

 (4/4) Host_RHEL.m9.u1.qcow2.ide.up.virtio_net.Guest.RHEL.8.7.0.x86_64.io-github-autotest-qemu.qemu_img_convert_from_vdi: PASS (1326.94 s)

**Comment by hellohellenmao** [comment]
@zhencliu Could you please help to review again? Thanks.

File: qemu/tests/cfg/qemu_img_convert_from_vdi.cfg [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+    only i440fx
+    no Host_RHEL.m6 Host_RHEL.m7 Host_RHEL.m8
+    required_qemu = [6.1.0-7, )
+    virt_test_type = qemu
+    type = qemu_img_convert_from_vdi
+    start_vm = no
+    kill_vm = yes
+    images = "rhel850"
+    image_name_rhel850 = images/rhel850
+    image_format_rhel850 = vdi
```
Comment (zhencliu) [inline qemu/tests/cfg/qemu_img_convert_from_vdi.cfg:12]: images = vdiimg
image_format_vdiimg = vdi

Set image_name_vdiimg in py file:
params['image_name_vdiimg'] = os.path.basename(params["vdi_image_url"]).split('.')[0]

vdi_image_address -> vdi_image_url

I am also OK with the current, thanks.

File: qemu/tests/qemu_img_convert_from_vdi.py [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+
+from provider import qemu_img_utils as img_utils
+
+
+def run(test, params, env):
+    """
+    1. Convert to a target image
+    2. Boot up a guest from the target file
+    3. Check the md5 value of the temporary file, and make sure the values are
+    the same
```
Comment (XueqiangWei) [inline qemu/tests/qemu_img_convert_from_vdi.py:15]: Please update the doc string, not found check md5 value related codes.
e.g.
1. Convert from vdi os image
2. Boot up from the target image
3. Check the data integrity

**Comment by XueqiangWei** [comment]
> 1. Convert from vdi os image
> 2. Boot up from the target image
> 3. Check the data integrity
> 
> Signed-off-by: Tingting Mao [timao@redhat.com](mailto:timao@redhat.com)
> 
> ID: 2112262

\"3. Check the data integrity\" is not in commit message, if we don't need it, please remove it, if we need it, please add it into commit message. Thanks.

File: qemu/tests/cfg/qemu_img_convert_from_vdi.cfg (hunk 6 above)
Comment (XueqiangWei) [reply qemu/tests/cfg/qemu_img_convert_from_vdi.cfg:12]: I agree with Zhenchao, rhel850 in the private cfg is OK.

Comment (hellohellenmao) [reply qemu/tests/cfg/qemu_img_convert_from_vdi.cfg:12]: Thanks.

File: qemu/tests/qemu_img_convert_from_vdi.py (hunk 7 above)
Comment (hellohellenmao) [reply qemu/tests/qemu_img_convert_from_vdi.py]: No need this step, and updated. Thanks.

**Comment by hellohellenmao** [comment]
> > 1. Convert from vdi os image
> > 2. Boot up from the target image
> > 3. Check the data integrity
> > 
> > Signed-off-by: Tingting Mao [timao@redhat.com](mailto:timao@redhat.com)
> > ID: 2112262
> 
> "3. Check the data integrity" is not in commit message, if we don't need it, please remove it, if we need it, please add it into commit message. Thanks.

Updated. Thanks.

**Comment by XueqiangWei** [comment]
(1/4) Host_RHEL.m9.u1.qcow2.ide.up.virtio_net.Guest.RHEL.8.7.0.x86_64.io-github-autotest-qemu.vdi_image_convert.to_raw: STARTED
 (1/4) Host_RHEL.m9.u1.qcow2.ide.up.virtio_net.Guest.RHEL.8.7.0.x86_64.io-github-autotest-qemu.vdi_image_convert.to_raw: PASS (13.12 s)
 (2/4) Host_RHEL.m9.u1.qcow2.ide.up.virtio_net.Guest.RHEL.8.7.0.x86_64.io-github-autotest-qemu.vdi_image_convert.to_vdi: STARTED
 (2/4) Host_RHEL.m9.u1.qcow2.ide.up.virtio_net.Guest.RHEL.8.7.0.x86_64.io-github-autotest-qemu.vdi_image_convert.to_vdi: PASS (8.28 s)
 (3/4) Host_RHEL.m9.u1.qcow2.ide.up.virtio_net.Guest.RHEL.8.7.0.x86_64.io-github-autotest-qemu.vdi_image_create: STARTED
 (3/4) Host_RHEL.m9.u1.qcow2.ide.up.virtio_net.Guest.RHEL.8.7.0.x86_64.io-github-autotest-qemu.vdi_image_create: PASS (3.82 s)
 (4/4) Host_RHEL.m9.u1.qcow2.ide.up.virtio_net.Guest.RHEL.8.7.0.x86_64.io-github-autotest-qemu.qemu_img_convert_from_vdi: STARTED
 (4/4) Host_RHEL.m9.u1.qcow2.ide.up.virtio_net.Guest.RHEL.8.7.0.x86_64.io-github-autotest-qemu.qemu_img_convert_from_vdi: PASS (1529.69 s)

**Comment by XueqiangWei** [comment]
LGTM,
Acked-by: xuwei@redhat.com

**Comment by zhencliu** [comment]
LGTM

**Pull request merged**